sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -495,6 +573,92 @@ case class Aggregate(
       super.statistics
     }
   }
+
+  private def sameOutput(output1: Seq[Attribute], output2: Seq[Attribute]): Boolean =
+    output1.size == output2.size &&
+      output1.zip(output2).forall(pair => pair._1.semanticEquals(pair._2))
```

Comment:
Think `forall { case (left, right) => left semanticEquals right }` (or with dots) could be more readable. WDYT?